src/test/scala/com/eclipsesource/schema/SchemaValidatorSpec.scala

```diff
@@ -165,39 +167,37 @@ class SchemaValidatorSpec extends PlaySpecification {
       }
     }
 
-  "should resolve references on the classpath via UrlResolver" in {
-    val validator = SchemaValidator().addUrlResolver(ClasspathUrlResolver())
+  "should resolve references on the classpath with ClasspathUrlResolver" in {
+    val validator = SchemaValidator().addUrlResolver(ClasspathUrlProtocolHandler())
     // some.json references location.json within JAR
     val someJson = getClass.getResourceAsStream("/some.json")
     val schema = JsonSource.schemaFromStream(someJson)
     val result = validator.validate(schema.get, Json.obj("location" -> Json.obj("name" -> "Munich")))
     result.isSuccess must beTrue
   }
 
-  "should resolve references on the classpath via UrlResolver and custom protocols (#65)" in {
+  "should resolve relative references on the classpath with relative URL handler (#65)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
-      .shouldResolveRelativeRefsWithCustomProtocols(true)
-    // some.json references location.json within JAR
+      .addRelativeUrlHandler(new ClasspathHandler)
+    // some-issue-65.json references location.json within JAR
     val someJson = getClass.getResourceAsStream("/some-issue-65.json")
     val schema = JsonSource.schemaFromStream(someJson)
     val result = validator.validate(schema.get, Json.obj("location" -> Json.obj("name" -> "Munich")))
     result.isSuccess must beTrue
   }
 
-  "should resolve relative references with custom protocols (#65)" in {
+  "should resolve relative references on the classpath with relative URL handler (#65 - 2)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
-      .shouldResolveRelativeRefsWithCustomProtocols(true)
+      .addRelativeUrlHandler(new ClasspathHandler)
     val mySchemaJson = getClass.getResourceAsStream("/my-schema.schema")
     val mySchema = JsonSource.schemaFromStream(mySchemaJson)
     val result = validator.validate(mySchema.get, Json.obj("foo" -> Json.obj("bar" -> "whatever")))
     result.isSuccess must beTrue
   }
 
-  "should fail resolving relative references with custom protocols if option not set(#65)" in {
+  "should fail resolving relative references without any relative URL handler available (#65)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
+      .addUrlResolver(ClasspathUrlProtocolHandler())
```

Comment:
Ok, I'm trying to re-cap this (I'm ignoring the case for protocol-less absolute URL handlers in the following, which is related to the issue about extracting the default behaviour mentioned earlier):

* a `UrlProtocolHandler` expects that the value it receives is a custom, absolute URL including the protocol, e.g. `classpath:///foo/bar.json`. The engine just passes the value of the ref to any absolute URL handler capable of handling the ref and we are basically done, right?
* a relative URL handler can have two flavours. The first one is, where we explicitly state the protocol to be used (?), e.g. `classpath://bar.json` and the second one is the protocol-less variant: `bar.json`. Two questions about this:

The first one is rather about relative handlers in general: what absolute URL are you expecting the engine to build here? Are your referring to the current resolution scope or perhaps to the physical location of the base schema?

The second question is about the protocol-ful variant of the relative classpath handler: we'd need to somehow encode the info for which protocol the handler should be triggered, so basically we'd end up with the same interface definition as for `UrlProtocolHandler` again (overloading `addRelativeUrlHandler` would be an option then), otherwise we'd not have the option to fail in case there's another relative URL handler that is able to resolve a given path, but has been registered for another protocol, correct? This is also why I said earlier that's there's currently no difference between the protocol-full and protocol-less variant of the relative handler because, right now, that info is missing. 
In the second case (`bar.json`) it's clear to me, that we only need a `URLStreamHandler` in the signature and that all protocol-less handlers are tried one after another until eventually one succeeds.

Did I get this right? About the open questions: maybe you could try to illustrate  those with some examples in order to avoid any ambiguity.